x-pack/plugins/apm/server/lib/service_nodes/index.ts

```diff
@@ -28,6 +29,14 @@ const getServiceNodes = async ({
 
   const params = mergeProjection(projection, {
     body: {
+      query: {
+        bool: {
+          filter: [
+            ...projection.body.query.bool.filter,
+            { exists: { field: SERVICE_NODE_NAME } },
```

Comment:
@dgieselaar do you think it makes sense to filter out metrics without `service.node.name`?

Reply:
No, because we still need to display metrics for services that do not have service.node.name set.

Reply:
and WDYT about adding a filter to only return data that has at least one of the statistics?
```js
.filter(
      (item) =>
        item.cpu || item.heapMemory || item.nonHeapMemory || item.threadCount
    );
```

Reply:
IMO that's fine. Ideally we can filter out the data when querying but that would be pretty complicated given the many possibilities of how these metrics are stored.